### PR TITLE
Drop support for Python 3.3

### PR DESCRIPTION
**Pull request by hugovk**
I found and removed a few more bits of code that supported Python 3.3, and updated some versions in docs.

This is against your req-py34 branch and merging would update your PR against the main repo: https://github.com/ipython/ipython/pull/10833.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -98,9 +98,9 @@ def is_stable(extra):
 rst_prolog += """
 .. important::
 
-    This documentation covers IPython versions 6.0 and higher. Beginning with
-    version 6.0, IPython stopped supporting compatibility with Python versions
-    lower than 3.3 including all versions of Python 2.7.
+    This documentation covers IPython versions 6.3 and higher. Beginning with
+    version 6.3, IPython stopped supporting compatibility with Python versions
+    lower than 3.4 including all versions of Python 2.7.
```
Comment (takluyver) [inline docs/source/conf.py]: I think we can leave this sentence alone - the important point is for Python 2 users, not for the relatively small number of people still on Python 3.3.

**Review by takluyver — COMMENTED** [review]
Thanks, this mostly looks fine, but can I ask you to do it with your editor configured not to strip trailing whitespace, so it's a smaller diff? Unnecessary whitespace changes can produce merge conflicts, and make tools like `git blame` less useful.

**Comment by hugovk** [comment]
Updated!

**Comment by takluyver** [comment]
Thanks :-)

**Pull request merged**